src/cryptoadvance/specter/specter.py

```diff
@@ -549,13 +560,26 @@ def add_new_user_otp(self, otp_dict):
         self.config["new_user_otps"].append(otp_dict)
         self._save()
 
-    def burn_new_user_otp(self, otp):
-        """ validates an OTP for user registration and removes it if valid"""
+    def validate_new_user_otp(self, otp):
+        """ validates an OTP for user registration and removes it if expired"""
+        if "new_user_otps" not in self.config:
+            return False
+        now = time.time()
+        for i, otp_dict in enumerate(self.config["new_user_otps"]):
+            if otp_dict["otp"] == otp:
+                if "expiry" in otp_dict and otp_dict["expiry"] > now:
```

Comment:
```suggestion
                if "expiry" in otp_dict and otp_dict["expiry"] < now and otp_dict["expiry"] > 0:
```
The condition here seems to be reversed. Also lacking the case where expiry is 0.

Reply:
done